projects/packages/waf/src/class-waf-rules-manager.php

```diff
@@ -167,18 +187,22 @@ private static function wrap_require( $required_file, $return_code = 'return;' )
 	/**
 	 * Generates the rules.php script
 	 *
-	 * @throws \Exception If file writing fails.
+	 * @global \WP_Filesystem_Base $wp_filesystem WordPress filesystem abstraction.
+	 *
+	 * @throws Waf_Exception         If filesystem is unavailable.
+	 * @throws File_System_Exception If file writing fails initializing rule files.
+	 * @throws File_System_Exception If file writing fails writing to the rules entrypoint file.
+	 *
 	 * @return void
 	 */
 	public static function generate_rules() {
-		/**
-		 * WordPress filesystem abstraction.
-		 *
-		 * @var \WP_Filesystem_Base $wp_filesystem
-		 */
 		global $wp_filesystem;
 
-		Waf_Runner::initialize_filesystem();
+		try {
+			Waf_Runner::initialize_filesystem();
+		} catch ( Waf_Exception $e ) {
+			throw $e;
+		}
```

Comment:
```suggestion
		Waf_Runner::initialize_filesystem();
```

We could also just not have the try-catch block here I think?

Reply:
Resolved via https://github.com/Automattic/jetpack/commit/c1d3effd93c8fd23355781bca6b324383c18d498